src/TsTypes.ts

```diff
@@ -58,54 +61,99 @@ export namespace TsType {
   export interface TsProp {
     name: string
     required: boolean
-    type: TsType
+    type: TsTypeBase
   }
 
-  export class Any extends TsType {
+  export class Any extends TsTypeBase {
     _type() {
       return 'any'
     }
   }
-  export class String extends TsType {
+  export class String extends TsTypeBase {
     _type() {
       return 'string'
     }
   }
-  export class Boolean extends TsType {
+  export class Boolean extends TsTypeBase {
     _type() {
       return 'boolean'
     }
   }
-  export class Number extends TsType {
+  export class Number extends TsTypeBase {
     _type() {
       return 'number'
     }
   }
-  export class Object extends TsType {
+  export class Object extends TsTypeBase {
     _type() {
       return 'Object'
     }
   }
-  export class Void extends TsType {
+  export class Void extends TsTypeBase {
     _type() {
       return 'void'
     }
   }
-  export class Literal extends TsType {
+  export class Literal extends TsTypeBase {
     constructor(private value: any) { super() }
     _type() {
       return this.value
     }
   }
 
-  export class Array extends TsType {
-    constructor(private type?: TsType) { super() }
+  export class EnumValue {
+    identifier: string
+    value: string
+
+    constructor(enumValues: string[]) {
+      let hasValue = !!enumValues[0]
```

Comment:
That's fine, but this package should support what TS supports. Ts supports `enum a {''}`, so should we.

If you want to avoid empty string enums, I recommend a tslint rule.